### PR TITLE
[IMP] {hr,sale}_timesheet: improve views in timesheet app

### PR DESCRIPTION
**Pull request by kcv-odoo**
Purpose of this PR to improve generic UX of timesheet app.

So in this PR done following changes:
- make readonly_timesheet field to add it in timesheet views to make
field readonly and avoid many xpath and simply override that method's
compute to change readonly condition.
- make so_line field default show in timesheet tree view and task form
view's timesheet page tree view.

task-2918757

### CODE REVIEW

**Review by auon-odoo — APPROVED** [review]
Looks good to me. Thank you :fire:

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left one comment, I think it will be easier if we have that hook to avoid doing many times filtered in the compute method of `readonly_timesheet` field. 🙂

File: addons/hr_timesheet/models/hr_timesheet.py [hunk 1]
```diff
@@ -83,6 +84,10 @@ def _get_display_name(project_id, task_id):
         timesheet_dict = {res['id']: _get_display_name(res['project_id'], res['task_id']) for res in timesheets_read}
         return list({**dict(result), **timesheet_dict}.items())
 
+    def _compute_readonly_timesheet(self):
+        # is overridden in other timesheet related modules
+        self.readonly_timesheet = False
```
[the remark this answers is not in the document]
Comment (kcv-odoo) [reply addons/hr_timesheet/models/hr_timesheet.py]: Done! Thank you :slightly_smiling_face:

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

robodoo r+

**Comment by xavierbol** [comment]
The review of Framework JS team is no longer needed since no files changed are in their scope, sorry about that, probably a wrong rebase. 🙂

**Comment by kcv-odoo** [comment]
Hey @xavierbol,

Re-based the branch as it was showing some conflicts :slightly_smiling_face:

**Review by xavierbol — APPROVED** [review]
Thanks. 🙂 

robodoo r+

**Pull request closed without merging**